docs/content/developers/design-keuzes.md

```diff
@@ -391,3 +391,66 @@ Dit kan door het toevoegen van de laatste status (`STATUSTYPE` met het hoogste
 Er is echter behoefte aan een consistente manier om zaken af te sluiten. In
 deze oplossing worden zowel `ZAAK.einddatum` als `STATUSTYPE` gebruikt waarbij
 er geen onduidelijkheid meer ontstaat over hoe een `ZAAK` wordt afgesloten.
+
+## Mechanisme voor autorisatie
+
+De DSO schrijft voor om OAUTH2 te gebruiken op API niveau voor authenticatie
+en autorisatie. Kort samengevat is OAUTH2 te complex voor onze behoeften en
+maken we geen gebruik van de features ervan. We gebruiken JSON Web Tokens
+om autorisatie te regelen, en dan in de HMAC 'signed' vorm (dus geen
+assymetrische encryptie).
```

Comment:
Waarom is gekozen voor HMAC 'signed vorm' en niet voor asymmetrische encryptie?

Reply:
Waarom worden de JWT tokens überhaupt gesigneerd? Immers binnen de gemeentelijke organisatie zou je de uitwisseling van tokens moeten kunnen vertrouwen en als je tokens uitwisselt met een andere organisatie heb je de bescherming van NLX. Heb je dan sowieso nog JWT nodig of kunnen de scopes gewoon als open tekst worden gecommuniceerd? Immers NLX draagt zorg voor betrouwbaarheid op mutual TLS niveau.

Reply:
> Waarom is gekozen voor HMAC 'signed vorm' en niet voor asymmetrische encryptie?

Uitgangspunt is dat er geen gevoelige gegevens in de payload zitten. Bij het gebruik van NLX willen we ook de HTTP headers loggen voor elke request. Het auditten van NLX logs laat ons dan toe om de payload te controleren in het JWT bij de request. Bij assymetrische encryptie zouden we hiervoor de public key nodig hebben om het te kunnen lezen, met HMAC signing niet.

> Waarom worden de JWT tokens überhaupt gesigneerd? 

Stel dat ik een medewerker ben binnen de organisatie die wat van techniek kent. Doordat ik binnen het interne netwerk zit, heb ik toegang tot de NLX outway van de organisatie, dus ik kan bij de zaken-APIs. Vervolgens ga ik zaakgegevens opvragen, en ik maak een token met payload die zegt dat ik alle scopes heb. Dit is problematisch - enkel bevoegde/gekende applicaties mogen valide tokens uitgeven. Dit beveilig je door de applicaties de tokens te laten signen.

De applicatie zelf weet (of kan opvragen) dat de betreffende medewerker niet alle scopes heeft, mogelijks mag die zelfs helemaal niets met zaken, dus deze moet misbruik stoppen.

De API accepteert enkel valide tokens, dus een medewerker die _spoofing_ doet, krijgt een fout van de API.

Reply:
Goeie vraag overigens, ik stelde die mezelf ook een tijdje terug :+1: